test/performance/operf.go

```diff
@@ -0,0 +1,538 @@
+	}
+	client := &http.Client{Transport: tr}
+
+	params := url.Values{}
+	params.Add("start", strconv.FormatInt(start, 10))
+	params.Add("end", strconv.FormatInt(end, 10))
+	params.Add("step", "6")
+	params.Add("namespace", *testNamespace)
+	params.Add("query", query)
+	params.Add("timeout", "30s")
+	body := strings.NewReader("")
+
+	metricsQuery := fmt.Sprintf("%s/api/v1/query_range?%s", prometheusURL, params.Encode())
+	req, err := http.NewRequest("GET", metricsQuery, body)
+
+	if err != nil {
+		fmt.Printf("Error querying Prometheus: %s", err)
+	}
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", *token))
+
+	resp, err := client.Do(req)
+	if err != nil {
+		fmt.Printf("Error in request: %s", err)
+		return ""
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		fmt.Println("Error in request:", resp.Status)
+		if resp.StatusCode == 403 {
+			fmt.Println("Probably token expired, renew token and try to execute again providing the new token.")
+		}
+		return ""
+	}
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		fmt.Println("Error reading response from ", req.URL, ":", err)
+	}
+
+	return string(data)
+}
+
+func (unit *lvmUnitMetrics) renderTemplate(metricsTemplate string) string {
+	tmpl, err := template.New("test").Parse(metricsTemplate)
+	if err != nil {
+		panic(err)
+	}
+	buf := new(bytes.Buffer)
+	err = tmpl.Execute(buf, unit)
+	if err != nil {
+		panic(err)
+	}
+	return buf.String()
+}
+
+func (unit *lvmUnitMetrics) getCPUMetrics(start int64, end int64) {
+	query := unit.renderTemplate(cpuQueryTemplate)
+	metrics := unit.getMetrics(start, end, query)
+	if metrics == "" {
+		panic("CPU Metrics not found")
+	}
+	unit.cpu.Max, unit.cpu.Avg = unit.getValues(metrics)
+}
+
+func (unit *lvmUnitMetrics) getRAMMetrics(start int64, end int64) {
+	query := unit.renderTemplate(memQueryTemplate)
+	metrics := unit.getMetrics(start, end, query)
+	if metrics == "" {
+		panic("RAM Metrics not found")
+	}
+	max, avg := unit.getValues(metrics)
+	unit.ram.Max = max / Mebibyte
+	unit.ram.Avg = avg / Mebibyte
+}
+
+func (unit *lvmUnitMetrics) getFSMetrics(start int64, end int64) {
+	query := unit.renderTemplate(filesystemTemplate)
+	metrics := unit.getMetrics(start, end, query)
+	if metrics == "" {
+		panic("FS Metrics not found")
+	}
+	max, avg := unit.getValues(metrics)
+	unit.fs.Max = max / Mebibyte
+	unit.fs.Avg = avg / Mebibyte
+}
+
+func (unit *lvmUnitMetrics) getAllMetrics(start int64, end int64) {
+	unit.start = time.Unix(start, 0)
+	unit.end = time.Unix(end, 0)
+	unit.getCPUMetrics(start, end)
+	unit.getRAMMetrics(start, end)
+	unit.getFSMetrics(start, end)
+
+}
+
+func (unit *lvmUnitMetrics) printMetrics(header bool) {
+	fmt.Println(strings.Repeat("-", 80))
+	fmt.Printf("Report for %s %s between %s and %s\n", unit.Name, unit.WorkloadType, unit.start, unit.end)
+	fmt.Printf("      CPU (max|avg) seconds: % 10.4f |  % 10.4f\n", unit.cpu.Max, unit.cpu.Avg)
```

Comment:
```suggestion
	fmt.Printf("\t CPU (max|avg) seconds: % 10.4f |  % 10.4f\n", unit.cpu.Max, unit.cpu.Avg)
```

Reply:
Using `\t` decrease the readability of a code in my opinion in such scenarios. Even though I have not written this code I still agree with this approach.

Do you see any loopholes in using spaces instead of tabs?

Reply:
Using the spaces directly might work too, but I believe they add a bit of ambiguity (As we might add/remove a single space and it will ruin the alignment). Similarly to how for a new line instead of using spaces and writing in the next line, `\n` is preferred.
Also, `\t` does not change the readability for the user; 
Thoughts?

Reply:
I think it is about the preference of an individual. I like the space more than `\t` and `\n` in the case of a new line obviously.

anyway changed it for now.